lib/pure/strutils.nim

```diff
@@ -154,6 +154,107 @@ proc isDigit*(c: char): bool {.noSideEffect,
     doAssert isDigit('8') == true
   return c in Digits
 
+proc isNumeric*(s: string): bool =
+  ## Checks whether the string is numeric.
+  ## When the string is an integer, float or exponential, it returns true,
+  ## otherwise it returns false.
+  ##
+  ## **Note:** The reason that `parseFloat()` is not used to achieve this is its
+  ## poor performance.
+  runnableExamples:
+    doAssert isNumeric("123") == true
+    doAssert isNumeric("123.45") == true
+    doAssert isNumeric("+123.45") == true
+    doAssert isNumeric("123.45e-2") == true
+    doAssert isNumeric("+123.45E-2") == true
+    doAssert isNumeric("-123.45e2") == true
+    doAssert not isNumeric("e123.45") == true
+    doAssert not isNumeric("abc") == true
+    doAssert not isNumeric("123abc") == true
+    doAssert not isNumeric("123.45.6") == true
+    doAssert not isNumeric("123.45e++5") == true
+    doAssert not isNumeric("5.2+e1") == true
+    doAssert isNumeric(".9") == true
```

Comment:
Personally don't think it's a good idea to treat these numbers as numbers (by default).

Reply:
I have added the parameter 'enabledNan' to indicate whether special forms of values such as' NaN ', 'Inf', '.9 ', '9.' are valid.